src/gpodder/gtkui/main.py

```diff
@@ -251,7 +251,8 @@ def changed_cb(spinbutton):
             diff = time.time() - self.config.software_update.last_check
             if diff > (60 * 60 * 24) * self.config.software_update.interval:
                 self.config.software_update.last_check = int(time.time())
-                self.check_for_updates(silent=True)
+                if not os.path.exists(gpodder.no_update_check_file):
```

Comment:
Would it be cleaner if this check was made earlier?
The results of time diff calculation does not affect this check
so it would be possible to short circuit earlier.

Of course this is a matter of style,
and gpodder is not my project in any way,
so it may be that I am just used to different conventions.

Reply:
The time check is a fast calculation that only succeeds once every 5 days, unless the user changes it. Where as the file check must access a potentially slow storage device, and increase startup time.

Reply:
Makes sense, being fast at runtime is more important than tiny readability gains.